org.elysium.parent/org.elysium.ui/src/org/elysium/ui/refactoring/LilyPondRefactoring.java

```diff
@@ -191,16 +229,38 @@ public Change perform(IProgressMonitor pm) throws CoreException {
 			if(sibling instanceof IFile){
 				IFile siblingFile=(IFile)sibling;
 				if(!compiledFilesRefactorTargets.contains(siblingFile) && support.isCompiled(siblingFile)){
-					boolean siblingCompiledFromSource=source.getFullPath().removeFileExtension().equals(siblingFile.getFullPath().removeFileExtension());
-					if(siblingCompiledFromSource){
-						result.add((IFile) sibling);
+					if(isCompiledFrom(source, siblingFile)){
+						result.add(siblingFile);
 					}
 				}
 			}
 		}
 		return result;
 	}
 
+	private boolean isCompiledFrom(IFile source, IFile compiled) {
+		if(source.getFullPath().removeFileExtension().equals(compiled.getFullPath().removeFileExtension())) {
+			return true;
+		} else if(LilyPondConstants.AUDIO_EXTENSION.equals(compiled.getFileExtension()) && 
```

Comment:
Sorry for not mentioning it earlier, but one source file can be compiled not only to multiple audio files, but also to multiple score files. See http://lilypond.org/doc/v2.19/Documentation/notation/multiple-output-files-from-one-input-file.

Reply:
Could we handle this case in a separate issue? When changing my local example to this use case, I noticed that in this case the numbering pattern for the midi files goes one level deeper: score-1.pdf has score-1-1.midi and score-1-2.midi...